lib/generate.sh

```diff
@@ -122,8 +140,52 @@ function _generate_sysext() {
     erofs)
       mkfs.erofs "${fname}" "${basedir}"
       ;;
-
   esac
+}
+# --
+
+function _generate_sysext() {
+  local extname="$1"
+  local basedir="$2"
+  local format="$3"
+  local ext_fs_size="$4"
+  local fname="$5"
+  local signed_ddi="$6"
+
+  if [[ ${signed_ddi} == true ]] ; then
+    extname="${extname}-signed-ddi"
+  fi
+
+  announce "Creating extension image '${fname}' and generating SHA256SUM"
+  (
+    export SYSTEMD_REPART_MKFS_OPTIONS_BTRFS="--mixed -m single -d single --shrink"
+    export SYSTEMD_REPART_MKFS_OPTIONS_EXT2="-E root_owner=0:0"
+    export SYSTEMD_REPART_MKFS_OPTIONS_EXT4="$SYSTEMD_REPART_MKFS_OPTIONS_EXT2"
+    export SYSTEMD_REPART_MKFS_OPTIONS_SQUASHFS="-all-root -noappend -xattrs-exclude '^btrfs.'"
```

Comment:
This needs EROFS, too, and the BTRFS compression options, available when rebasing this branch